### ISSUE TITLE
Explicit delimiter option

### ISSUE THREAD

**Issue by mdshw5** [arq5x/bedtools2#138]
This might be considered "feature creep", but sometimes it would be useful to allow a delimiter other than '\t', such as a space. A simple `--delim` option could allow users to specify such a delimiter instead of making a new, properly formatted file.

**Comment by arq5x** [comment]
@nkindlon - thoughts on this?

**Comment by nkindlon** [comment]
Sure. The current type checker (FileRecordTypeChecker) doesn't assume the delimiter is a tab. It actually uses a method I patched together to test tabs, commas, and I think semil-colons to see which one appears a consistent number of times in the first 4 lines of the file and decides that one is the delimiter. Use of a --delim could simply bypass that step.

**Comment by arq5x** [comment]
I guess one issue is cases where the A file and the B file(s) have different delimiters. This complexity alone makes me think this is leaning a bit too far into the feature creep realm...

**Comment by mdshw5** [comment]
Would it be reasonable to add spaces as a case to `FileRecordTypeChecker`?

**Comment by mdshw5** [comment]
It looks like delimiters are tested in hierarchical order, so adding spaces last would not break any current workflows.

**Comment by nkindlon** [comment]
Good points, both. Matt, my concern with adding spaces to the typeChecker is that it could break for long VCF records and the like that can contain many words in a long string in the optional fields. That's part of why it was never added in the first place. Spaces can also appear in the names and chrom names fields for standard formats, or really anywhere in the non-standard formats that BedPlus supports. As none of that is predictable, I just don't think it's a safe idea. Note that the delimiter has to be established before we can make any determination of what files we're dealing with, or even how many fields it has, so separate rules for different formats wouldn't be possible.

Aaron, I'd later also thought about the A and B files, as well as multiple databases that might not have the same delim. We could go as far as having a -delimA and -delimB options, with the latter behaving like the col ops in map, i.e the user can give one entry to apply to all databases, or one for each database. Realize all of this is extremely easy internally, as all files have their own typeChecker. It's a matter of deciding what behavior we want on the command line. 

I think for now we should just add the --delim option, identical to the linux cut command, as we try to mimic the behaviour of standard linux tools. That will at least support Matt's initial request. The onus will be on the user to ensure that that's correct for all files involved. They can always use sed to replace spaces with tabs, or vice-versa. If that proves inadequate, we can try a more robust approach as described in a future release. Sound good?

**Comment by mdshw5** [comment]
Sounds fine to me, but note that I don't even remember why I opened this request, so don't rush or risk breaking anything. Thanks.

**Comment by nkindlon** [comment]
No prob. That's a pretty easy one. We're doing small bug fixes and enhancements this week, so that'll be out soon.

**Comment by nkindlon** [comment]
Just realized map (as well as merge) already have a -delim option, so I'll have to call this -parse_delim to avoid confusing it with that. I'd rather not change the existing options for a new one, as this will likley be used less.

**Comment by arq5x** [comment]
Let us hold off on this one until we can give some more thought. Lower priority in my opinion.

**Comment by nkindlon** [comment]
Ok.
